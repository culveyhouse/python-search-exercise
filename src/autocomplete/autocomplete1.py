class Autocomplete1:
    def __init__(self, suggestions):
        self.suggestions = suggestions

    def performSearch(self, term):
        """Searches through the possible suggestions, creating an array of
        matches for any books whose title matches the specified search term.

        This should perform a case sensitive contains search.

        Example:
            With 'biology' specified as the term a book with the title
            'Microbiology' should be returned. A book with the title 'Biology'
            should not be returned.

        Args:
            term - the term to search for in book titles.

        Return:
            List of results where the title contains the search term.
        """
        # TODO: Implement this.
        return []
