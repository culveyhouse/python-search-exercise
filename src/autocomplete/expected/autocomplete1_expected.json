{
    "attribution": "Marjorie Kelly Cowan",
    "briefDescription": "",
    "chapterPrice": 7.99,
    "copyrightYear": 2011,
    "description": "Get to know the smallest of organisms up close and personal with Cowan\u2019s Microbiology on Inkling. Spin dozens of 3D molecules on their axes, watch more than 80 animations of various processes and explore more than 100 enhanced figures, all exclusive to Inkling.",
    "edition": "3rd edition",
    "price": 129.99,
    "primaryCategory": "Science",
    "publisher": "McGraw-Hill",
    "subCategories": [],
    "subtitle": "A Systems Approach",
    "thumbnail": "data/thumbnails/book-10.jpeg",
    "title": "Microbiology",
    "url": "https://www.inkling.com/store/book/microbiology-marjorie-kelly-cowan-3rd/"
}
